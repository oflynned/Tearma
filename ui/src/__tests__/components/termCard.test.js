import React from "react";
import renderer from "react-test-renderer";

import TermCard from "../../components/termCard";

describe("<TermCard />", () => {
	it("should render correctly", () => {
		const result = {
			en: {
				term: "en term",
				domains: ["domain1", "domain2"]
			},
			ga: {
				term: "ga term",
				gender: "masculine",
				declension: 1,
				mutations: {
					nominativeSingular: "nom sing",
					nominativePlural: "nom plu",
					genitiveSingular: "gen sing",
					genitivePlural: "gen plu"
				},
				domains: ["domain1", "domain2"]
			}
		};

		const component = renderer.create(<TermCard result={result} />).toJSON();
		expect(component).toMatchSnapshot();
	});
});
