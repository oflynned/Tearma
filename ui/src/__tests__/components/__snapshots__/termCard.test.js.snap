// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TermCard /> should render correctly 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 card"
>
  <div
    className="contents"
  >
    <div>
      <h5
        className="MuiTypography-root-29 MuiTypography-h5-45 header"
        style={
          Object {
            "fontSize": "18px",
          }
        }
      >
        Ga term
         / 
        En term
      </h5>
    </div>
    <div
      className="margin-content"
    >
      <div
        className="MuiChip-root-64 chip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        style={
          Object {
            "fontSize": "14px",
          }
        }
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-83"
        >
          noun
        </span>
      </div>
      <div
        className="MuiChip-root-64 chip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        style={
          Object {
            "fontSize": "14px",
          }
        }
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-83"
        >
          masculine
        </span>
      </div>
      <div
        className="MuiChip-root-64 chip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        style={
          Object {
            "fontSize": "14px",
          }
        }
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-83"
        >
          1st declension
        </span>
      </div>
      <div
        className="MuiChip-root-64 chip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        style={
          Object {
            "fontSize": "14px",
          }
        }
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-83"
        >
          domain1
        </span>
      </div>
      <div
        className="MuiChip-root-64 chip"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        style={
          Object {
            "fontSize": "14px",
          }
        }
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-83"
        >
          domain2
        </span>
      </div>
    </div>
    <div
      className="margin-content"
    >
      <div
        className="MuiGrid-container-89 MuiGrid-spacing-xs-16-112"
      >
        <div
          className="MuiGrid-item-90 MuiGrid-grid-xs-6-123"
        >
          <h5
            className="MuiTypography-root-29 MuiTypography-h5-45"
            style={
              Object {
                "fontSize": "16px",
              }
            }
          >
            nom sing
          </h5>
          <h6
            className="MuiTypography-root-29 MuiTypography-subtitle1-47 MuiTypography-colorTextSecondary-62"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            nominative singular
          </h6>
        </div>
        <div
          className="MuiGrid-item-90 MuiGrid-grid-xs-6-123"
        >
          <h5
            className="MuiTypography-root-29 MuiTypography-h5-45"
            style={
              Object {
                "fontSize": "16px",
              }
            }
          >
            nom plu
          </h5>
          <h6
            className="MuiTypography-root-29 MuiTypography-subtitle1-47 MuiTypography-colorTextSecondary-62"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            nominative plural
          </h6>
        </div>
      </div>
      <div
        className="MuiGrid-container-89 MuiGrid-spacing-xs-16-112"
      >
        <div
          className="MuiGrid-item-90 MuiGrid-grid-xs-6-123"
        >
          <h5
            className="MuiTypography-root-29 MuiTypography-h5-45"
            style={
              Object {
                "fontSize": "16px",
              }
            }
          >
            gen sing
          </h5>
          <h6
            className="MuiTypography-root-29 MuiTypography-subtitle1-47 MuiTypography-colorTextSecondary-62"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            genitive singular
          </h6>
        </div>
        <div
          className="MuiGrid-item-90 MuiGrid-grid-xs-6-123"
        >
          <h5
            className="MuiTypography-root-29 MuiTypography-h5-45"
            style={
              Object {
                "fontSize": "16px",
              }
            }
          >
            gen plu
          </h5>
          <h6
            className="MuiTypography-root-29 MuiTypography-subtitle1-47 MuiTypography-colorTextSecondary-62"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            genitive plural
          </h6>
        </div>
      </div>
    </div>
    <div
      className="margin-content"
    >
      <p
        className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-colorTextSecondary-62"
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        Related terms: administration
      </p>
    </div>
  </div>
</div>
`;
